### PR TITLE
pretty_print will use #inspect if a subclass redefines it

### PR DESCRIPTION
**Pull request by notEthan**
per discussion at https://github.com/rails/rails/pull/15172#issuecomment-69445752

that discussion wasn't entirely conclusive, but this may be desirable. 

@matthewd @mwesigwa

### CODE REVIEW

**Comment by mwesigwa** [comment]
@notEthan this commit will both restore the original inspect behavior and perform your pretty_print behavior in absence of a user_defined inspect method. This is exactly what I was asking for :+1:

**Comment by imanel** [comment]
I don't think this is good idea. If user want to overwrite `inspect` then it should affect only it, and not `pretty_print`. I'm aware that this is fix for pry's logic of "show pp instead of inspect", but then you should overwrite `pretty_print` method in your code instead of `inspect`.

Please don't break others code just because your favorite debugging tool does not behave as it should.

**Comment by rafaelfranca** [comment]
@imanel this is how Ruby pretty_print works. https://github.com/ruby/ruby/blob/trunk/lib/pp.rb#L301-303

File: activerecord/lib/active_record/core.rb [hunk 1]
```diff
@@ -456,22 +456,26 @@ def inspect
...
-          pp.seplist(column_names, proc { pp.text ',' }) do |column_name|
-            column_value = read_attribute(column_name)
-            pp.breakable ' '
-            pp.group(1) do
-              pp.text column_name
-              pp.text ':'
-              pp.breakable
-              pp.pp column_value
+      if self.class.instance_method(:inspect).owner != ActiveRecord::Base.instance_method(:inspect).owner
+        pp.text inspect
```
Comment (rafaelfranca) [inline activerecord/lib/active_record/core.rb]: Can we just call `super`?

**Comment by sgrif** [comment]
I made some style changes in the merge commit, updated to call `super` and changed the code to use a guard clause to better preserve the blame history of the method.

**Comment by notEthan** [comment]
thanks @sgrif

**Pull request merged**